MediaBrowser.Controller/Entities/TV/Series.cs

```diff
@@ -213,6 +212,24 @@ public List<BaseItem> GetSeasons(User user, DtoOptions options)
             return LibraryManager.GetItemList(query);
         }
 
+        public Dictionary<int, string> GetSeasonNames()
+        {
+            if (_seasonNames.Count > 0)
+            {
+                return _seasonNames;
+            }
+
+            return Children.OfType<Season>()
```

Comment:
Shouldn't this merge into the dictionary instead?

Reply:
I think that's an even better solution. Will deal with it later